smallrye-reactive-messaging-aws-sns/src/test/java/io/smallrye/reactive/messaging/aws/sns/AwsSnsTest.java

```diff
@@ -0,0 +1,91 @@
+package io.smallrye.reactive.messaging.aws.sns;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.eclipse.microprofile.reactive.streams.operators.SubscriberBuilder;
+import org.jboss.weld.environment.se.Weld;
+import org.jboss.weld.environment.se.WeldContainer;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.reactivestreams.Subscriber;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.reactivex.Flowable;
+
+public class AwsSnsTest extends AwsSnsTestBase {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AwsSnsTest.class);
+    private WeldContainer container;
+    String topic = "sns-test";
+    String appUrl = "http://docker.for.mac.host.internal";
```

Comment:
We should use TestContainer Rules to start the fake SNS automatically.
See https://www.testcontainers.org/quickstart/junit_4_quickstart/, and https://www.testcontainers.org/features/creating_container/

Reply:
Looks interesting 👍 , will check this out.